networking_nsxv3/api/rpc.py

```diff
@@ -108,90 +81,46 @@ class NSXv3ServerRpcApi(object):
 
     rpc_version = nsxv3_constants.NSXV3_SERVER_RPC_VERSION
 
-    _LIMIT = 100
-    _CREATE_AFTER = datetime.utcfromtimestamp(0).isoformat()
-
-    def __init__(self, context, topic, host):
-        target = oslo_messaging.Target(topic=topic, version=self.rpc_version)
+    def __init__(self):
+        target = oslo_messaging.Target(topic=nsxv3_constants.NSXV3_SERVER_RPC_TOPIC, 
+                                       version=self.rpc_version)
+        self.context = neutron_context.get_admin_context()
         self.client = rpc.get_client(target)
-        self.context = context
-        self.host = host
-        self.topic = topic
+        self.host = cfg.CONF.host        
 
     @log_helpers.log_method_call
-    def get_port_revision_tuples(
-            self, limit=_LIMIT, created_after=_CREATE_AFTER):
+    def get_ports_revisions(self, limit, offset):
         cctxt = self.client.prepare()
-        return cctxt.call(
-            self.context,
-            'get_port_revision_tuples',
-            host=self.host,
-            limit=limit,
-            created_after=created_after)
+        return cctxt.call(self.context, 'get_ports_revisions',
+                          host=self.host, limit=limit, offset=offset)
 
     @log_helpers.log_method_call
-    def get_qos_policy_revision_tuples(
-            self, limit=_LIMIT, created_after=_CREATE_AFTER):
+    def get_qoses_revisions(self, limit, offset):
         cctxt = self.client.prepare()
-        return cctxt.call(self.context, 'get_qos_policy_revision_tuples',
-                          limit=limit, created_after=created_after)
```

Comment:
What is the motivation to go from "created_after/limit" to "offset/limit"?
"Offset" queries are usually less performant and instable over modifications.

Reply:
It was mainly due to the performance (thought that offset has better performance) and unification (every query can have limit and offset, but not true for create_after, or at least for some it does not make sense).

I see your point and it makes sense. Will roll that back.

- [ ] Replace offset with create after

Reply:
That is still open, correct?

Reply:
- [x] Improve DB query precision replacing offset with created_after